### PR TITLE
Add concurrency control to connector validate_filtering method

### PR DESCRIPTION
**Pull request by wangch079**
## Closes https://github.com/elastic/enterprise-search-team/issues/4161

This PR adds concurrency control to the `validate_filtering` method of `Connector` class

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

File: connectors/byoc.py [hunk 1]
```diff
@@ -627,7 +630,12 @@ async def validate_filtering(self, validator):
                 if validation_result.state == FilteringValidationState.VALID:
                     filter_["active"] = filter_.get("draft")
 
-        await self.index.update(doc_id=self.id, doc={"filtering": filtering})
+        await self.index.update(
```
Comment (timgrein) [inline connectors/byoc.py:633]: I'm wondering if we could provide a method inside the client class itself handling sequence numbers. To me personally this feels more as a concern of the client and not the connector, but just thinking out loud. So we're having something like `update_with_concurrency_control` and the API looks the same as `update`, but uses concurrency control under the hood.

**Review by timgrein — APPROVED** [review]
Only one question, if we should put a method somewhere else, but not blocking. LGTM!

File: connectors/byoc.py (hunk 1 above)
Comment (wangch079) [reply connectors/byoc.py:633]: >I'm wondering if we could provide a method inside the client class itself handling sequence numbers.

Did you mean the [ESIndex](https://github.com/elastic/connectors-python/blob/db83cac4a66e0687e3741da80475e6d0692ca738/connectors/es/index.py#L18) class?

> So we're having something like `update_with_concurrency_control` and the API looks the same as `update`, but uses concurrency control under the hood.

Could you elaborate?

Do you mean we should add this method to [ESIndex](https://github.com/elastic/connectors-python/blob/db83cac4a66e0687e3741da80475e6d0692ca738/connectors/es/index.py#L18) class?

```python
async def update_with_concurrency_control(self, es_document, doc):
    return await self.client.update(
        index=self.index_name,
        id=es_document.id,
        doc=doc,
        if_seq_no=es_document._seq_no,
        if_primary_term=es_document._primary_term,
    )
```

Comment (wangch079) [reply connectors/byoc.py:633]: Hi @timgrein , I will merge this PR first, we can make a separate PR if needed.

**Pull request merged**